articles/support-vector-machines/index.md

```diff
@@ -0,0 +1,443 @@
+## Understanding support vector machines
+
+### Introduction
+Support vector machines are machine learning methods used mostly to solve classification problems. We can also use them to solve regression and outlier detection problems. Support vector machines perform well even with large datasets with a large number of features. They also don't consume a lot of memory since they only use a subset of the training data in determining the decision function.
+
+Support vector machines separate the dataset into classes using the  **maximum margin hyperplane(MMH)**.
+
+### Prerequisites
+To follow along, you are required to have an understanding of linear algebra, numpy, and scikit-learn.
+ **Definition of terms**
+ 1. **Support Vectors**: They are the data points closest to the hyperplane that helps decide the maximum margin. 
+ 2. **Hyperplane**: It's defined as the plane that separates the data into different classes.
+ 3. **Margin**: This is the path or gap between two lines separating different classes.
+ 
+
+The following diagram gives a visual representation of the support vector machine concept.
+
+ ![terms](support-vector-machines.jpg)
+
+***mathematical formulation***
+
+Given training vectors `𝑥`<sub>i</sub> `∈ R`<sup>n</sup>, `i=1,. . . , n`, in two classes, and a vector `𝑦 ∈ {1,−1},` our goal is to find `𝑤 ∈ R`<sup>`n`</sup>*** and `𝑏 ∈ R`  such that `w` is normal to the hyperplanes.  (*see figure below*)
+ 
+ ![enter image description here](vector.jpg)
+
+ `w.u>=c` ; ie. project `u` on the normal
+
+`c = -b`
+
+if `w.u + b >=0` ; then `u` is then classified as positive (decision rule).
+
+`w.x + b >=1`
+
+`w.x + b <=-1`
+
+introducing  `y`<sub>i</sub> such that `y`<sub>i</sub>`=+-1` for +ve positive samples and -ve for negative samples.
+
+*for the negative class* 
+
+*replacing y with `-1` in the equation we have* 
+
+`-(w.x + b) <=-1`
+
+*dividing both sides by `-1` we have*
+
+`w.x + b >= 1`
+
+and hence the equations turn out to be the same,
+
+`y`<sub>i</sub>`(w.x + b ) >=1`, for +ve samples
+
+`y' <sub>i</sub>`(w.x +b)>=1`, for -ve samples and so we can use one of them can be used as the decision function ie.
+
+`y(w.x + b)-1>=0`
+
+Introducing a constraint to the above equation so that whenever we have a sample on the gutter (support vectors), it should then give us 0.
+
+So we use:
+
+`y`<sub>i</sub>`(w.x + b)-1 = 0` as our contstraint.
+
+Now that we have both the decision function and its constraint we, therefore, need to find the width of the margin. We have:
+
+***width***  = `(x`<sub>i</sub> `- x ) = 2/`<sub>||`w`||</sub> as  our optimization problem.(*see fegure below*)
```

Comment:
These equations aren't written correctly. Use \frac{}{} to represent fractions correctly,